hack/testing/rsyslog/roles/rsyslog/defaults/main.yaml

```diff
@@ -0,0 +1,1509 @@
+                           else "False" }}'
+
+# .. envvar:: rsyslog__message_reduction
+#
+# Enable or disable message reduction. This is disabled by default so that log
+# parsers like :program:`fail2ban` can work correctly.
+rsyslog__message_reduction: False
+
+
+# rpm packages
+# adding rsyslog__logging_packages
+# ------------------
+
+# .. envvar:: rsyslog__base_packages
+#
+# List of default rpm packages to install.
+rsyslog__base_packages: [ 'rsyslog' ]
+
+# .. envvar:: rsyslog__logging_packages
+#
+# List of rpm packages for Common Logging.
+rsyslog__viaq_prereq_packages: [ 'nmap-ncat', 'systemd-python', 'policycoreutils', 'checkpolicy', 'policycoreutils-python' ]
+rsyslog__viaq_packages: [ 'rsyslog-elasticsearch', 'rsyslog-mmjsonparse', 'rsyslog-mmkubernetes', 'rsyslog-mmnormalize', 'libfastjson', 'liblognorm' ]
+
+# .. envvar:: rsyslog__tls_packages
+#
+# List of rpm packages required for TLS support.
+rsyslog__tls_packages: [ 'rsyslog-gnutls', 'ca-certificates' ]
+
+# .. envvar:: rsyslog__packages
+#
+# List of additional rpm packages to install.
+rsyslog__packages: []
+
+
+# User, group, home, file ownership
+# ---------------------------------
+
+# .. envvar:: rsyslog__user
+#
+# The unprivileged system user account used by the ``rsyslogd`` daemon, when
+# unprivileged operation is enabled.
+rsyslog__user: '{{ "syslog" if rsyslog__unprivileged|bool else "root" }}'
+
+# .. envvar:: rsyslog__group
+#
+# The unprivileged system group account used by the ``rsyslogd`` daemon, when
+# unprivileged operation is enabled.
+rsyslog__group: '{{ "syslog" if rsyslog__unprivileged|bool else "root" }}'
+
+# .. envvar:: rsyslog__home
+#
+# The home directory of the :envvar:`rsyslog__user` user, dependent on the OS
+# defaults. Takes effect only when the unprivileged mode is enabled.
+rsyslog__home: '{{ "/var/log" }}'
+
+# .. envvar:: rsyslog__file_owner
+#
+# The account which will be set as the owner of the log files generated by
+# ``rsyslogd``. When an unprivileged mode is enabled, it should be the same as
+# the ``rsyslogd`` user account.
+rsyslog__file_owner: '{{ rsyslog__user }}'
+
+# .. envvar:: rsyslog__file_group
+#
+# The system group which will be set as the default group of the log files
+# generated by ``rsyslogd``.
+rsyslog__file_group: 'adm'
+
+# .. envvar:: rsyslog__default_logfiles
+#
+# (??? -- need logrotate???)
+# audit/audit.log, etc.?
+# List of default logfiles rsyslog role. This list is used in the logrotate_
+# configuration to create log rotation, as well as to change the file owner/
+# group when unprivileged operation is enabled.
+rsyslog__default_logfiles:
+  - '/var/log/syslog'
+  - '/var/log/kern.log'
+  - '/var/log/auth.log'
+  - '/var/log/user.log'
+  - '/var/log/daemon.log'
+  - '/var/log/messages'
+  - '/var/log/mail.log'
+  - '/var/log/mail.info'
+  - '/var/log/mail.warn'
+  - '/var/log/mail.err'
+  - '/var/log/cron.log'
+  - '/var/log/lpr.log'
+  - '/var/log/debug'
```

Comment:
this looks very debian specific

Reply:
@richm, rsyslog__default_logfiles is used in logrotate, which I haven't looked in yet.  (currently, commented out...)  Do we want to support it in this ansible work?

Reply:
I don't think so - the rsyslog rpm configures logrotate by default

Reply:
@nhosoi see https://src.fedoraproject.org/rpms/rsyslog/tree/f28 - the logrotate config file is rsyslog.log

Reply:
@nhosoi also note that rhel 7 and rhel 8 rsyslog are different than this

Reply:
@richm, yes, it looks rhel7 expects /etc/logrotate.d/syslog and rhel8 (and new fedoras) do /etc/logrotate.d/rsyslog.
```
$ rpm -ql rsyslog | egrep logrotate
/etc/logrotate.d/rsyslog
```
The contents look identical...
```
/var/log/cron
/var/log/maillog
/var/log/messages
/var/log/secure
/var/log/spooler
{
    missingok
    sharedscripts
    postrotate
	/bin/kill -HUP `cat /var/run/syslogd.pid 2> /dev/null` 2> /dev/null || true
    endscript
}
```
This ansible role is hoping to add:
```
          notifempty
          delaycompress
          compress
```
We could add them to the syslog or rsyslog whichever exists on the host?

Reply:
> We could add them to the syslog or rsyslog whichever exists on the host?

I don't think so.  For now, the rsyslog rpm should be the source of truth for rsyslog.conf and logrotate config for rsyslog.  If anything, by default, deploying the rsyslog playbook should configure the system _exactly_ as it is currently deployed by the rpm.

Reply:
Ok.  I'm convinced that logrotate should not be touched in this ansible deployment...  I'm removing the logrotate related code from the tasks.